code/modules/projectiles/pins.dm

```diff
@@ -229,3 +226,16 @@
 	if(gun)
 		gun.pin = null
 	return ..()
+
+//Station Locked
+
+/obj/item/firing_pin/away
+	name = "station locked pin"
+	desc = "A firing pin that only will fire when off the station."
+
+/obj/item/firing_pin/away/pin_auth(mob/living/user)
```

Comment:
make sure it respects emagging if it doesn't already, just checking.

Reply:
It does, or should like any normal emagged firing pin.